File: scripts/fix-admin-fetch.js
#!/usr/bin/env node
// Fix all adminFetch callers that do `const json = await res.json()` pattern
// Since adminFetch now returns parsed JSON directly, we need to inline it.

const fs = require('fs');
const path = require('path');

const srcDir = path.join(__dirname, '..', 'src', 'app');

function processFile(filePath) {
  let content = fs.readFileSync(filePath, 'utf8');
  let changed = false;
  
  // Pattern 1: const res = await adminFetch(...);\n      const json = await res.json();
  // Replace with: const json = await adminFetch(...);
  const pattern1 = /const (res|json|data|result) = await adminFetch\(([^)]*)\);\s*\n(\s*)const (\w+) = await \1\.json\(\);/g;
  if (pattern1.test(content)) {
    content = content.replace(pattern1, (match, var1, args, indent, var2) => {
      return `const ${var2} = await adminFetch(${args});`;
    });
    changed = true;
  }

  // Pattern 2: When the first var is used for status check like `if (!res.ok)` before `.json()`
  // const res = await adminFetch(...);\n      if (!res.ok) ...\n      const json = await res.json();
  // This is trickier - for now skip these
  
  if (changed) {
    fs.writeFileSync(filePath, content, 'utf8');
    console.log('Fixed:', filePath);
  }
}

function walkDir(dir) {
  const entries = fs.readdirSync(dir, { withFileTypes: true });
  for (const entry of entries) {
    const fullPath = path.join(dir, entry.name);
    if (entry.isDirectory()) {
      walkDir(fullPath);
    } else if (entry.name.endsWith('.tsx') || entry.name.endsWith('.ts')) {
      processFile(fullPath);
    }
  }
}

walkDir(srcDir);
console.log('Done!');

File: scripts/fix-admin-fetch.py
#!/usr/bin/env python3
"""Fix remaining adminFetch + .json() patterns robustly."""
import re, os, glob

pattern = re.compile(
    r'(const\s+(res|json|data|result)\s*=\s*await\s+adminFetch\([^)]*\);)\s*\n(\s*)(const\s+\w+\s*=\s*await\s+\2\.json\(\);)',
    re.MULTILINE
)

count = 0
for f in glob.glob('src/app/**/*.tsx', recursive=True) + glob.glob('src/app/**/*.ts', recursive=True):
    with open(f, 'r') as fh:
        content = fh.read()
    new_content = pattern.sub(r'\3\4', content)
    if new_content != content:
        with open(f, 'w') as fh:
            fh.write(new_content)
        count += 1
        print(f'Fixed: {f}')

print(f'\nTotal files fixed: {count}')